src/mappers/mapper34.asm

```diff
@@ -0,0 +1,78 @@
+// Mapper 34: BNROM
+
+scope Mapper34: {
+Init:
+  addi sp, 8
+  sw ra, -8(sp)
+
+// 32K
+  jal TLB.AllocateVaddr
+  lui a0, 0x1'0000 >> 16  // align 64K to leave a 32k guard page unmapped
+
+  ls_gp(sw a0, mapper34_prgrom_vaddr)
+  ls_gp(sb a1, mapper34_prgrom_tlb_index)
+
+// 0x8000-0x1'0000
+  addi t0, a0, -0x8000
+  la_gp(t1, Write)
+  lli t2, 0
+  lli t3, 0x80
+
+-
+  sw t0, cpu_read_map + 0x80 * 4 (t2)
+  sw t1, cpu_write_map + 0x80 * 4 (t2)
+  addi t3, -1
+  bnez t3,-
+  addi t2, 4
+
+// Hard wired CHR mapping 0x0000-0x2000 (8K)
+  ls_gp(lw t0, chrrom_start)
+  lli t1, 8
+  lli t2, 0
+-
+  sw t0, ppu_map (t2)
+  addi t1, -1
+  bnez t1,-
+  addi t2, 4
+
+// Initially map PRG ROM bank 0, Nametable 0
+  j Write_alt
+  lli cpu_t0, 0
+
+Write:
+// cpu_t0: data
+// cpu_t1: address (unused)
+  addi sp, 8
+  sw ra, -8(sp)
+
+Write_alt:
+  // and t0, cpu_t0, 3
```

Comment:
Remove

Reply:
This was meant to point specifically at line 49, sorry, I'm not very experienced with GitHub reviews.

Reply:
I think you commented on the line correctly! will remove